File: OOP/seriale/seriale/Series.cpp
#include "Series.h"


Series::Series(string iTitle, string iGenre, float iScore, int iSeasons) : Production(iTitle, iGenre, iScore)
{
	seasons = iSeasons;
}

int Series::getSeasons()
{
	return seasons;
}

void Series::setSeasons(int s)
{
	seasons = s;
}

string Series::stringify()
{
	string result = Production::stringify();
	result += to_string(getSeasons());
	return result;
}


Series::~Series()
{
}

File: serialeJava/src/Series.java
import java.util.ArrayList;
import java.util.Date;

class Series extends Product {

    private int numberOfSeasons;
    private ArrayList<Season> seasons;
    private String genre;

    Series(String imgPath, String title, String description, Date productionDate, int duration, Distributor distributor, ArrayList<String> countries, float score, int numberOfSeasons, ArrayList<Season> seasons, String genre) {
        super(imgPath, title, description, productionDate, duration, distributor, countries, score);
        this.numberOfSeasons = numberOfSeasons;
        this.seasons = seasons;
        this.genre = genre;
    }

    @Override
    public String toString() {
        String poprzedni = super.toString();
        return poprzedni + "\nSeries{" +
                "numberOfSeasons=" + numberOfSeasons +
                ", seasons=" + seasons +
                ", genre='" + genre + '\'' +
                '}';
    }

    public int getNumberOfSeasons() {
        return numberOfSeasons;
    }

    public void setNumberOfSeasons(int numberOfSeasons) {
        this.numberOfSeasons = numberOfSeasons;
    }

    public ArrayList<Season> getSeasons() {
        return seasons;
    }

    public void setSeasons(ArrayList<Season> seasons) {
        this.seasons = seasons;
    }

    public String getGenre() {
        return genre;
    }

}
